ssg/build_remediations.py

```diff
@@ -146,6 +151,81 @@ def get_populate_replacement(remediation_type, text):
     sys.exit(1)
 
 
+def from_file(file_path, env_yaml):
+    mod_file = []
+    config = defaultdict(lambda: None)
+
+    fix_file_lines = jinja_process_file(file_path, env_yaml).splitlines()
+
+    # Assignment automatically escapes shell characters for XML
+    for line in fix_file_lines:
+        line += "\n"
+        if line.startswith('#'):
+            try:
+                (key, value) = line.strip('#').split('=')
+                if key.strip() in ['complexity', 'disruption',
+                                   'platform', 'reboot',
+                                   'strategy']:
+                    config[key.strip()] = value.strip()
+                else:
+                    if not line.startswith(FILE_GENERATED):
+                        mod_file.append(line)
+            except ValueError:
+                if not line.startswith(FILE_GENERATED):
+                    mod_file.append(line)
+        else:
+            mod_file.append(line)
+
+    return mod_file, config
+
+
+def process_fix(fixes, args, env_yaml, product, file_path, fix_name):
+    if not is_supported_filename(args.remediation_type, file_path):
+        return
+
+    fix_contents, config = from_file(file_path, env_yaml)
+
+    if is_applicable_for_product(config['platform'], product):
+        fixes[fix_name] = (fix_contents, config)
+    elif not config['platform']:
+        raise RuntimeError(
+            "The '%s' remediation script does not contain the "
+            "platform identifier!" % (file_path))
+
+
+def write_fixes(args, fixes):
+    fixcontent = ElementTree.Element("fix-content", system="urn:xccdf:fix:script:sh",
+                                     xmlns="http://checklists.nist.gov/xccdf/1.1")
+    fixgroup = get_fixgroup_for_type(fixcontent, args.remediation_type)
+
+    remediation_functions = get_available_functions(args.build_dir)
+
+    for fix_name in fixes:
+        fix_contents, config = fixes[fix_name]
+
+        fix_elm = ElementTree.SubElement(fixgroup, "fix")
+
+        fix_elm.set("rule", fix_name)
+
+        if config['complexity']:
+            fix_elm.set("complexity", config['complexity'])
+        if config['disruption']:
+            fix_elm.set("disruption", config['disruption'])
+        if config['reboot']:
+            fix_elm.set("reboot", config['reboot'])
+        if config['strategy']:
+            fix_elm.set("strategy", config['strategy'])
+
+        fix_elm.text = "".join(fix_contents)
```

Comment:
What about joining over a newline in exchange of not appending newline to those strings in `from_file`?

Reply:
Sure. One quirk is that `fix_elm.text` needs to end with a new line due to `expand_xccdf_subs`. Don't ask me why but yeah:

```
Error performing XCCDF <sub> substitution for function package_remove in package_nss-pam-ldapd_removed fix. Exiting...
```